app/models/concerns/anonymizable.rb

```diff
@@ -1,21 +1,38 @@
 module Anonymizable
+  extend ActiveSupport::Concern
+
   def anonymize_personal_data_columns!
-    personal_data_columns = self.class.columns.select do |column|
-      column.name.in?(self.class.personal_data_column_names)
-    end
+    update_columns(self.class.anonymized_attributes) # rubocop:disable Rails/SkipsModelValidations
+  end
 
-    anonymized_attributes = personal_data_columns.to_h do |column|
-      [column.name, anonymous_value(column)]
+  class_methods do
+    def anonymize_all!(scope)
+      raise "L'anonymisation en masse est désactivée en production pour éviter les catastrophes" if Rails.env.production?
+
+      scope.update_all(anonymized_attributes) # rubocop:disable Rails/SkipsModelValidations
     end
 
-    update_columns(anonymized_attributes) # rubocop:disable Rails/SkipsModelValidations
-  end
+    def anonymized_attributes
+      personal_data_columns = columns.select do |column|
+        column.name.in?(personal_data_column_names)
+      end
 
-  private
+      personal_data_columns.to_h do |column|
+        [column.name, anonymous_value(column)]
+      end
+    end
 
-  def anonymous_value(column)
-    if column.type.in?(%i[string text])
-      "[valeur anonymisée]"
+    def anonymous_value(column)
+      case column.type.to_sym
+      when :string, :text
+        "[valeur anonymisée]"
+      when :date
+        column.null ? nil : Date.new(1900, 1, 1)
+      when :integer
+        column.null ? nil : 0
+      else
+        raise "Don't know how to anonymize column #{column.name} (#{column.type})"
```

Comment:
si la colonne est nullable, est-ce que ça ne vaut pas le coup de la passer à null ?

Reply:
Après réflexion, je partirais bien sur

```ruby
      if column.type.in?(%i[string text])
        "[valeur anonymisée]"
      else
        column.default
      end
```

Comme ça, si la colonne est NOT NULL mais ne définit pas de valeur par défaut, Postgres pète et on ajoute un cas spécifique dans cette méthode si nécessaire. :woman_dancing: